### PR TITLE
  Preliminar complete pinout definition 

### PR DESCRIPTION
**Pull request by cua-cua**
Pinout proposal with last constraints.

### CODE REVIEW

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -42,21 +42,38 @@ on both TI1 and TI2 edges.
...
-PB12  GPIO       Left motor driver IN1
-PB13  GPIO       Left motor driver IN2
-PB14  GPIO       Right motor driver IN2
-PB15  GPIO       Right motor driver IN1
+====  =========  ======  ==============================
+Pin   Function   Type	 Connected to
+====  =========  ======  ==============================
+PA0   ADC12_IN0  Input   Battery 1
+PA1   ADC12_IN1	 Input	 Battery 2
+PA2   ADC12_IN2	 Input 	 Gyroscope Vref
```
Comment (Peque) [inline docs/source/configuration.rst]: I'd use spaces here instead of tabs (for consistency). :wink:

File: docs/source/configuration.rst [hunk 2]
```diff
@@ -42,21 +42,38 @@ on both TI1 and TI2 edges.
...
+====  =========  ======  ==============================
+PA0   ADC12_IN0  Input   Battery 1
+PA1   ADC12_IN1	 Input	 Battery 2
+PA2   ADC12_IN2	 Input 	 Gyroscope Vref	
+PA3   ADC12_IN3  Input	 Gyroscope Vout
+PA4   ADC12_IN4  Input	 Sensor left collector      
+PA5   ADC12_IN5  Input	 Sensor front-left collector
+PA6   ADC12_IN6  Input	 Sensor front-right collector
+PA7   ADC12_IN7  Input	 Sensor right collector
+PA8   GPIO	 Output  Sensor emitter switch
```
Comment (Peque) [inline docs/source/configuration.rst]: Do you know how much time it takes for this GPIO to switch state?

File: docs/source/configuration.rst [hunk 3]
```diff
@@ -42,21 +42,38 @@ on both TI1 and TI2 edges.
...
+PA15  TIM2_CH1   Input   Left motor encoder, channel A
+PB0   TIM3_CH3   Output	 Left motor PWM
+PB1   TIM3_CH4   Output	 Right motor PWM
+PB3   TIM2_CH2   Input	 Left motor encoder, channel B
+PB4   GPIO       Output  Led  
+PB5   GPIO       Output  Led  
+PB6   TIM4_CH1   Input	 Right motor encoder, channel B
+PB7   TIM4_CH2   Input 	 Right motor encoder, channel A
+PB8   GPIO       Output  Led 
+PB9   GPIO       Output  Led
```
Comment (Peque) [inline docs/source/configuration.rst]: Maybe we could left these 2 free?

**Comment by Peque** [comment]
I guess #31 can be closed, right? (#46 seems to supersede #31)

Also, I like the #31 commit description better (i.e.: "Preliminar complete pinout definition"). :stuck_out_tongue_winking_eye:

File: docs/source/configuration.rst (hunk 2 above)
Comment (cua-cua) [reply docs/source/configuration.rst]: The maximum speed for output can be configured until 50 MHz.
From this blog: http://hertaville.com/stm32f0-gpio-tutorial-part-1.html , for F0 boards "the output pins fastest toggle speed is every two clock cycles", so I guess that for F1 will be similar.

File: docs/source/configuration.rst (hunk 1 above)
Comment (cua-cua) [reply docs/source/configuration.rst]: Right.

File: docs/source/configuration.rst (hunk 3 above)
Comment (cua-cua) [reply docs/source/configuration.rst]: Ok

File: docs/source/configuration.rst [hunk 4]
```diff
@@ -42,22 +42,39 @@ on both TI1 and TI2 edges.
...
-PB13  GPIO       Left motor driver IN2
-PB14  GPIO       Right motor driver IN2
-PB15  GPIO       Right motor driver IN1
-====  =========  ==============================
+====  =========  ======  ==============================
+Pin   Function   Type	   Connected to
+====  =========  ======  ==============================
+PA0   ADC12_IN0  Input   Battery 1
+PA1   ADC12_IN1	 Input   Battery 2
+PA2   ADC12_IN2	 Input   Gyroscope Vref
```
Comment (Peque) [inline docs/source/configuration.rst]: :sweat_smile:

Comment (cua-cua) [reply docs/source/configuration.rst]: Damn it!

**Pull request merged**